aoc/src/main/scala/Day03.scala

```diff
@@ -0,0 +1,69 @@
+import scala.io._
+
+object Day03 extends App:
+
+  val day: String =
+    this.getClass.getName.drop(3).init
+
+  val start1: Long =
+    System.currentTimeMillis
+
+  val input: List[String] =
+    Source
+      .fromResource(s"input$day.txt")
+      .getLines
+      .toList
+
+  def parser(s: String): (String,String) =
+    s.splitAt(s.length / 2)
+
+  def commonLetter(compartments: (String, String)): List[Char] =
+    val letters: List[String] = compartments._1.sliding(1).toList
+    for (
+      x <- letters;
+      y <- x
+      if compartments._2.contains(y)
+    ) yield y
+
+  def priority(list: List[Char]): List[Int] =
+    val alphabetLower = "abcdefghijklmnopqrstuvwxyz".zip(Stream.from(1)).toList
+    val alphabetUpper = "ABCDEFGHIJKLMNOPQRSTUVWXYZ".zip(Stream.from(27)).toList
+    val alphabet: List[(Char,Int)] = alphabetLower ++ alphabetUpper
+    for (
+      (x,y) <- alphabet;
+      z <- list
+      if z == x
+    ) yield y
+
+  def commonLetterGroup(compartments: List[String]): List[Char] =
```

Comment:
you could transform the `List[Char]`s to two `Set[Char]`s and employ the `s1.intersect(s2)` method to find the `Char`s the two lists have in common - combined with the trick to pattern match directly into the one common value after transforming the remaining intersection back into a `List[Char]` again, ie. for part one i wrote:

```val List(c) = left.toSet.intersect(right.toSet).toList```